### DISCUSSION TITLE
Converting HDR Images from .avif to .heif loses color profile

### DISCUSSION THREAD

**Discussion by borstl** [ImageMagick/ImageMagick#6621] [Help]
### ImageMagick version

7.1.1-15

### Operating system, version and so on

MacOS Ventura

### Description

Hello there,

when converting an 10-Bit .avif Rec. 2020 PQ image to .heif using `magick input.avif output.heif` I'm losing that radiance of the image and it looks flat. Even when using `magick input.avif -profile Rec2020-Rec1886.icc output.heif` it's looking not as flat, but is far off the radiant .avif version.
Did I missed something while converting or is magick not able to preserve the HDR effect of my images yet?

I assume that I need the right color profile from the .avif (Rec. 2020 PQ). The other one only got sRGB and Rec2020-Rec1886 as shown in `sips -g all output.heif` or Adobe Bridge.

I put the files in the attachment for better analysis. Filetypes are not supported here, so zip it is unfortunately.
[HDRImages.zip](https://github.com/ImageMagick/ImageMagick/files/12539135/HDRImages.zip)

**Comment by snibgo** [comment]
I don't know much about HDR or AVIF or HEIF. According to IM and exiftool, HDR.avif has no ICC color profile. It does contain XMP profile, which we can see with ...
```
magick HDR.avif xmp:x.txt
```
... and examine x.txt

If it had an ICC color profile, IM could convert it to any other ICC profile. As far as I know, IM does copy the HDR metadata in the XMP metadata, but does no processing with it.

If you think the pixel data is encoded with Rec2020-Rec1886.icc, you can (as you did) assign that profile, then either leave that data as it is, or convert to another profile.

**Reply by novomesk** [reply]
Color profile in AVIF can not only as ICC but also as NCLX box (set of enums which represent color primaries and transfer/gamma function).

```
avifdec -i HDR.avif 
Image decoded: HDR.avif
 * Resolution     : 2912x4368
 * Bit Depth      : 10
 * Format         : YUV444
 * Alpha          : Absent
 * Range          : Full
 * Color Primaries: 9
 * Transfer Char. : 16
 * Matrix Coeffs. : 9
 * ICC Profile    : Absent
 * XMP Metadata   : Present (12155 bytes)
 * Exif Metadata  : Present (650 bytes)
 * Transformations: None
 * Progressive    : Unavailable
 * CLLI           : 3248, 59
 * 1 timescales per second, 1.00 seconds (1 timescales), 1 frame
 * Frame:
   * Decoded frame [0] [pts 0.00 (0 timescales)] [duration 1.00 (1 timescales)] [2912x4368]
```

Color Primaries: 9 = BT2020

Transfer Char. : 16 = SMPTE2084 (PQ)

**Reply by borstl** [reply]
Thank you for your inputs. The part with the ICC seems to be a misdirection. I've learned these days that there are also workflows from ITU to convert for example between HLG and PQ and so on [source](https://www.color.org/hdr/02-Luke_Wallis.mp4) which the ICC does not seem to support yet.. maybe there is really something lost in conversion?

The only observable difference I can spot are the different profiles:
```bash
  sips -g all HDR.avif
  pixelWidth: 2912
  pixelHeight: 4368
  typeIdentifier: public.avif
  format: avif
  formatOptions: default
  dpiWidth: 72.000
  dpiHeight: 72.000
  samplesPerPixel: 3
  bitsPerSample: 10
  hasAlpha: no
  space: RGB
  profile: Rec. ITU-R BT.2100 PQ
  creation: 2023:09:06 16:39:34
  make: Canon
  model: Canon EOS 5D
  software: Adobe Photoshop Camera Raw 15.5.1 (Macintosh)
```

```bash
  sips -g all noProfilingOutput.heif
  pixelWidth: 2912
  pixelHeight: 4368
  typeIdentifier: public.heif
  format: heif
  formatOptions: default
  dpiWidth: 72.000
  dpiHeight: 72.000
  samplesPerPixel: 3
  bitsPerSample: 10
  hasAlpha: no
  space: RGB
  profile: sRGB IEC61966-2.1
  creation: 2023:09:06 16:39:34
  make: Canon
  model: Canon EOS 5D
  software: Adobe Photoshop Camera Raw 15.5.1 (Macintosh)
```
  
Unfortunately I can't find a profile online which I could assign to the conversion which would be 'Rec. ITU-R BT.2100 PQ' and the NCLX is also a new term for me where I couldn't find anything about it.

**Reply by novomesk** [reply]
NCLX structure in libheif header:

https://github.com/strukturag/libheif/blob/6ee6762657cfb0a921860d1a9d4cf36fa0eb5633/libheif/heif.h#L927

**Comment by snibgo** [comment]
> Unfortunately I can't find a profile online which I could assign to the conversion which would be 'Rec. ITU-R BT.2100 PQ'

Perhaps see https://w3c.github.io/png-hdr-pq/ , which contains a link to https://w3c.github.io/png-hdr-pq/icc/ITUR_2100_PQ_FULL.icc . There is a note that: "The gAMA chunk and embedded ICC profile specified in the table above are not accurate, and their processing is not required to reproduce the image as intended by the author."

**Comment by dandyweng** [comment]
I'm encountering the same issue when attempting to resize an AVIF HDR image that was exported from Lightroom, using the Rec. ITU-R BT.2100 PQ color profile. Despite trying the 'ITUR_2100_PQ_FULL.ICC' file, I'm still not achieving the correct results.

**Reply by borstl** [reply]
My current solution is to additionally use Photomator. It renders your photos with I guess [Apple's Metal API](https://developer.apple.com/videos/play/wwdc2023/10181) into HDR jpeg, heic, avif and png.

**Reply by BrianMwit** [reply]
Just had the same issue with resizing AVIF HDR PQ (not gain map). Got it working using `ffmpeg -i input.avif -vf scale=1080:-2 -crf 10 output.avif` instead
